File: scripts/test-calendar.js
require('dotenv').config();
const pool = require('../src/db/pool');
const { listUpcomingEvents, createEvent } = require('../src/services/calendar');

const USER_ID = 1;

async function main() {
  console.log('Fetching your upcoming events...\n');
  const events = await listUpcomingEvents(USER_ID, 10);

  if (events.length === 0) {
    console.log('No upcoming events found.');
  } else {
    for (const event of events) {
      const start = event.start.dateTime || event.start.date;
      console.log(`- ${event.summary}  (${start})`);
    }
  }

  console.log('\nCreating a test event one hour from now...');
  const { rows } = await pool.query('SELECT timezone FROM users WHERE id = $1', [USER_ID]);
  const timeZone = rows[0].timezone;

  const start = new Date(Date.now() + 60 * 60 * 1000);
  const end = new Date(start.getTime() + 30 * 60 * 1000);

  const created = await createEvent(USER_ID, {
    summary: 'Day Planner test event',
    description: 'Created by scripts/test-calendar.js to verify write access.',
    startDateTime: start.toISOString(),
    endDateTime: end.toISOString(),
    timeZone,
  });

  console.log(`Created: ${created.summary}`);
  console.log(`View it: ${created.htmlLink}`);

  await pool.end();
}

main().catch((err) => {
  console.error('Test script failed:', err.message);
  process.exit(1);
});

File: scripts/test_calendar.py
import sys
from datetime import datetime, timedelta
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parent.parent))

from app.db.base import SessionLocal  # noqa: E402
from app.db.models import User  # noqa: E402
from app.services.calendar import create_event, list_upcoming_events  # noqa: E402

USER_ID = 1


def main():
    db = SessionLocal()

    print("Fetching your upcoming events...\n")
    events = list_upcoming_events(db, USER_ID, max_results=10)

    if not events:
        print("No upcoming events found.")
    else:
        for event in events:
            start = event["start"].get("dateTime", event["start"].get("date"))
            print(f"- {event.get('summary', '(no title)')}  ({start})")

    print("\nCreating a test event one hour from now...")
    user = db.get(User, USER_ID)
    time_zone = user.timezone

    start = datetime.now().astimezone() + timedelta(hours=1)
    end = start + timedelta(minutes=30)

    created = create_event(
        db,
        USER_ID,
        summary="Day Planner test event (Python)",
        description="Created by scripts/test_calendar.py to verify write access.",
        start_datetime=start.isoformat(),
        end_datetime=end.isoformat(),
        time_zone=time_zone,
    )

    print(f"Created: {created['summary']}")
    print(f"View it: {created['htmlLink']}")

    db.close()


if __name__ == "__main__":
    main()
